app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,21 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    post = current_user.posts.new(post_params)
+    if post.save
+      redirect_to user_post_path(post.author.id, post.id), notice: 'Published successfully!'
+    else
+      flash.now[:alert] = 'Failed to publish post!'
+    end
+  end
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
```

Comment:
- [ ] Kindly make `post_params` private.
see reference for [strong params](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters)